lib/node_modules/@stdlib/ndarray/unshift/test/test.assign.js

```diff
@@ -0,0 +1,272 @@
+var unshift = require( './../lib/assign.js' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof unshift, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function throws an error if provided a first argument which is not an ndarray', function test( t ) {
+	var values;
+	var out;
+	var i;
+
+	out = zeros( [ 4 ] );
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		[],
+		{},
+		function noop() {}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[ i ] ), TypeError, 'throws an error when provided ' + values[ i ] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			unshift( value, 0.0, out );
+		};
+	}
+});
+
+tape( 'the function throws an error if provided a first argument which is an ndarray with more than one dimension', function test( t ) {
+	var values;
+	var out;
+	var i;
+
+	out = zeros( [ 4 ] );
+	values = [
+		zeros( [ 2, 2 ] ),
+		zeros( [ 3, 1, 2 ] ),
+		zeros( [ 2, 2, 2 ] ),
+		zeros( [ 3, 3, 3, 3 ] ),
+		zeros( [ 3, 1, 2 ] )
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[ i ] ), TypeError, 'throws an error when provided ' + values[ i ] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			unshift( value, 0.0, out );
+		};
+	}
+});
+
+tape( 'the function throws an error if provided an output argument which is not an ndarray', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		[],
+		{},
+		function noop() {}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[ i ] ), TypeError, 'throws an error when provided ' + values[ i ] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			unshift( zeros( [ 2 ]), 0.0, value );
+		};
+	}
+});
+
+tape( 'the function throws an error if provided an output argument which is an ndarray with more than one dimension', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		zeros( [ 2, 2 ] ),
+		zeros( [ 3, 1, 2 ] ),
+		zeros( [ 2, 2, 2 ] ),
+		zeros( [ 3, 3, 3, 3 ] ),
+		zeros( [ 3, 1, 2 ] )
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[ i ] ), TypeError, 'throws an error when provided ' + values[ i ] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			unshift( zeros( [ 2 ] ), 0.0, value );
+		};
+	}
+});
+
+tape( 'the function throws an error if provided insufficient arguments', function test( t ) {
+	t.throws( badValue, Error, 'throws an error' );
+	t.end();
+
+	function badValue() {
+		unshift( zeros( [ 2 ] ), zeros( [ 2 ] ) );
+	}
+});
+
+tape( 'the function prepends the provided provided scalar values to the input ndarray and assigns results to an output ndarray (row-major, contiguous)', function test( t ) {
```

Comment:
```suggestion
tape( 'the function prepends provided provided scalar values to the input ndarray and assigns results to an output ndarray (row-major, contiguous)', function test( t ) {
```

Applies here and elsewhere.